book/src/user/parameters.md

```diff
@@ -0,0 +1,88 @@
+}
+```
+
+Zebra uses the [bellman crate groth16 implementation](https://github.com/zkcrypto/bellman/blob/main/src/groth16/mod.rs) for all groth16 types.
+
+Each time a transaction has any sprout joinsplit, sapling spend or sapling output these loaded parameters will be used for the verification process. Zebra verifies in parallel and by batches, these parameters are used on each verification done.
+
+There are 2 different zebrad commands to get this parameters from the internet and load them into zebra:
+
+## zebrad download
+
+When this command is executed Zebra will create a path for each parameter file. For example, in linux, parameter files will be at:
+
+- `/home/$USER/.zcash-params/sapling-output.params`
+- `/home/$USER/.zcash-params/sapling-spend.params`
+- `/home/$USER/.zcash-params/sprout-groth16.params`
+
+This files are available for [download](https://download.z.cash/downloads/) and their hash for verification are part of the Zcash protocol: 
+
+https://zips.z.cash/protocol/protocol.pdf#grothparameters
+
+Zebra will next use the [zcash_proofs](https://github.com/zcash/librustzcash/tree/main/zcash_proofs) crate to call a function with the 3 paths as arguments.
+
+Function `zcash_proofs::load_parameters()` will take the 3 paths, check if the data is already there, verify the hashes and download the content if needed.
+
+```sh
+$ ./target/release/zebrad download
+2022-12-29T16:54:50.273253Z  INFO zebrad::components::tracing::component: started tracing component filter="info" TRACING_STATIC_MAX_LEVEL=LevelFilter::TRACE LOG_STATIC_MAX_LEVEL=Trace
+2022-12-29T16:54:50.274611Z  INFO {zebrad="535d0ad" net="Main"}: zebrad::application: initialized rayon thread pool for CPU-bound tasks num_threads=12
+2022-12-29T16:54:50.275486Z  INFO {zebrad="535d0ad" net="Main"}: zebrad::commands::download: checking if Zcash Sapling and Sprout parameters have been downloaded
+2022-12-29T16:54:50.312716Z  INFO {zebrad="535d0ad" net="Main"}: zebra_consensus::primitives::groth16::params: checking and loading Zcash Sapling and Sprout parameters
+2022-12-29T16:54:56.865148Z  INFO {zebrad="535d0ad" net="Main"}: zebra_consensus::primitives::groth16::params: Zcash Sapling and Sprout parameters downloaded and/or verified
+$
+```
+
+## zebrad start
+
+The alternative way is to let zebra do the above process at startup.
+
+Before Zebra attempts to verify any of the 3 mentioned circuits it needs to have the parameters in place. For this reason zebra start will check for the parameters and download them if needed each time it is started.
+
+At zebra startup, when initializing the verifiers, a separate task is created to do the same as the `zebra download` command. This allows Zebra sync to make progress before having the parameters. Note that these parameters are needed after Sapling activation which happens at block `419_200` in the Mainnet and block `15` in the Testnet. Zebra will panic if the parameters are not there when verifying those blocks and above.
```

Comment:
Can you show me where this panic happens?

I tried to design it so Zebra would wait for the download to finish, but I might have got it wrong.

```suggestion
At zebra startup, when initializing the verifiers, a separate task is created to do the same as the `zebra download` command. This allows Zebra sync to make progress before having the parameters. Note that these parameters are needed after Sapling activation which happens at block `419_200` in the Mainnet and block `15` in the Testnet. Zebra will wait for the parameters to download before verifying those blocks and above.
```